File: app/schemas/wallet.py
from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from app.models.wallet import WalletType


class WalletHoldingRead(BaseModel):
    address: str
    wallet_type: WalletType
    confidence_score: Decimal | None
    approximate_balance: Decimal
    rank: int
    scanned_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/wallet.ts
export interface WalletHolding {
  address: string;
  wallet_type: "unknown" | "whale" | "smart_money" | "vc" | "exchange" | "influencer" | "early_adopter";
  confidence_score: string | null;
  approximate_balance: string;
  rank: number;
  scanned_at: string;
}
